src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -142,9 +162,9 @@ private static void writeReadsSingle(
             return new Tuple2<>(gatkRead, samRecordWritable);
         });
 
-        // coalesce is needed for writing to a single bam.
+        // do a total sort so that all the reads in partition i are less than those in partition i+1
         final JavaPairRDD<GATKRead, SAMRecordWritable> out =
-                rddSamRecordWriteable.sortByKey(new ReadCoordinateComparator(header)).coalesce(1);
+                totalSortByKey(rddSamRecordWriteable, new ReadCoordinateComparator(header), GATKRead.class);
```

Comment:
@tomwhite I'm not sure I understand why a separate fn `totalSortByKey` is necessary.  Isn't this exactly what `sortByKey` is doing?  It looks it to me based on the code.  The main change appears to me to be to eliminate the `.coalesce(1)` and then perform the stitching as you described.